Sources/IR/Analysis/Module+AnalyzeProjection.swift

```diff
@@ -0,0 +1,149 @@
+import FrontEnd
+
+extension Module {
+
+  /// Check if projection `f` is well-formed, raising diagnostics to `log` as needed, and populates `projectionSkeletons` with
+  /// the relevant information for expanding projections.
+  ///
+  /// A well formed projection must have exactly one yield on each path through the function.
+  mutating func analyzeProjection(
+    _ f: Function.ID, reportingDiagnosticsTo log: inout DiagnosticSet
+  ) {
+    if !self[f].isSubscript {
+      return
+    }
+
+    if let p = try? self[f].analyzeProjection(reportingDiagnosticsTo: &log) {
+      projectionSkeletons[f] = p
+    }
+  }
+
+}
+
+extension Function {
+
+  /// Check if projection `self` is well-formed, raising diagnostics to `log` as needed,
+  /// and returns the relevant information for expanding projections.
+  fileprivate func analyzeProjection(
+    reportingDiagnosticsTo log: inout DiagnosticSet
+  ) throws -> ProjectionSkeleton {
+    // Gather all the yield points in the projection.
+    let yieldPoints: [InstructionID] = instructionIdentities.filter({ self[$0] is Yield })
+    let yieldBlocks = yieldPoints.map({ block(of: $0) })
```

Comment:
```suggestion
    let yieldBlocks = yieldPoints.map(block(of:_))
```
eta-expansion rocks!